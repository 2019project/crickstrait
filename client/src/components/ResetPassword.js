import React, { Component } from "react";
import { resetPassword, sendOtp, verifyOtp } from "../actions/Users";
import "../styles/ResetPassword.css";

import { connect } from "react-redux";

export class ResetPassword extends Component {
  state = {
    email: "",
    password: "",
    confirmPassword: "",
    otp: "",
    errors: {},
    show_otp_field: false,
    inputField: true
  };

  OnChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  // Send OTP
  sendOTP = e => {
    e.preventDefault();
    if (this.validateForm()) {
      let user = {
        email: this.state.email
      };
      this.props.sendOtp(user);
    }
    this.setState({
      show_otp_field: true
    });
  };

  // Reset Password
  onReset = () => {
    let user = {
      email: this.state.email,
      password: this.state.password
    };
    this.props.resetPassword(user);
  };

  // Verify OTP function
  verifyOTP = e => {
    e.preventDefault();
    let user = {
      otp: this.state.otp
    };
    if (this.props.verifyOtp(user)) {
      this.onReset();
      this.setState({
        email: "",
        password: "",
        confirmPassword: "",
        otp: ""
      });
    }
  };

  // Function for validating form input fields
  validateForm() {
    let errors = {};
    let formIsValid = true;
    if (!this.state.email) {
      formIsValid = false;
      errors["email"] = "*Please enter your Email-ID.";
    }

    if (!this.state.password) {
      formIsValid = false;
      errors["password"] = "*Please enter your Password.";
    }

    if (!this.state.confirmPassword) {
      formIsValid = false;
      errors["confirmPassword"] = "*Please Enter Confirm Password";
    }

    if (this.state.password != this.state.confirmPassword) {
      formIsValid = false;
      errors["confirmPassword"] = "*Password Don't Match";
    }

    this.setState({
      errors: errors
    });
    return formIsValid;
  }

  render() {
    return (
      <div>
        {/* Reset password main conatiner */}
        <div id="resetPasswordMainContainer">
          {/* reset password form */}
          <form id="resetPasswordForm">
            <div>
              <button id="resetButton">Reset Password</button>
            </div>
            <input
              id="resetPasswordInput"
              placeholder="Email"
              type="text"
              name="email"
              value={this.state.email}
              onChange={this.OnChange}
            />

            <p id="resetPasswordErrorMessage">{this.state.errors.email}</p>

            <input
              id="resetPasswordInput"
              placeholder="New Password"
              type="password"
              name="password"
              value={this.state.password}
              onChange={this.OnChange}
            />

            <p id="resetPasswordErrorMessage">{this.state.errors.password}</p>

            <input
              id="resetPasswordInput"
              placeholder="Confirm Password"
              type="password"
              name="confirmPassword"
              value={this.state.confirmPassword}
              onChange={this.OnChange}
            />

            <p id="resetPasswordErrorMessage">
              {this.state.errors.confirmPassword}
            </p>

            {/* Checking the state of otp filed , if the filed is true then show otp fileds */}
            {this.state.show_otp_field === true ? (
              <>
                <input
                  id="resetPasswordInput"
                  placeholder="Enter Otp"
                  type="password"
                  name="otp"
                  value={this.state.otp}
                  onChange={this.OnChange}
                />

                <p>{this.props.error ? this.props.error : "Hello"}</p>

                <p id="resetPasswordErrorMessage">{this.state.errors.otp}</p>

                <button onClick={this.verifyOTP} id="resetPasswordButton">
                  Reset
                </button>
              </>
            ) : (
              <button onClick={this.sendOTP} id="resetPasswordButton">
                Proceed
              </button>
            )}
          </form>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  users: state.usersReducer.users,
  error: state.usersReducer.error
});

export default connect(mapStateToProps, { resetPassword, sendOtp, verifyOtp })(
  ResetPassword
);
