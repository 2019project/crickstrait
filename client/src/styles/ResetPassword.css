#resetPasswordErrorMessage {
  color: #ff4d4d;
  font-size: 14px;
}
#resetPasswordForm {
  background: transparent;
}

#resetPasswordMainContainer {
  height: 483px;
  text-align: center;
  background-color: #2c3e50;
}

#resetPasswordInput {
  border: none;
  outline: none;
  background: transparent;
  color: #fff;
  border-bottom: 1px solid #cacaca;
  padding: 18px;
  width: 70%;
  font-size: 16px;
}

#resetPasswordInput:focus,
#resetPasswordInput:active {
  transform: scaleX(1.1);

  border-bottom: 2px solid #39abf7;
}

#resetPasswordButton {
  padding: 15px;
  background: transparent;
  outline: none;
  font-size: 16px;
  border: 3px solid #39abf7;
  color: #39abf7;
  margin-top: 40px;
  cursor: pointer;
}

#resetButton {
  background: transparent;
  color: #fff;
  outline: none;
  padding: 20px;
  font-size: 20px;
  border: none;
  border-bottom: 2px solid #39abf7;
  /* margin-bottom: 20px; */
}

@media screen and (min-width: 430px) {
  #resetPasswordMainContainer {
    height: 657px;
    text-align: center;
  }

  #resetPasswordInput {
    width: 25%;
  }

  #resetPasswordForm {
    height: auto;
    width: 100%;
  }
}
