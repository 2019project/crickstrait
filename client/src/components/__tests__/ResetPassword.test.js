import React from "react";
import { shallow, mount } from "enzyme";
import { ResetPassword } from "../ResetPassword";

const resetPassword = jest.fn();
const verifyOtp = jest.fn();

const wrapper = shallow(
  <ResetPassword resetPassword={resetPassword} verifyOtp={verifyOtp} />
);

describe("Testing ResetPassword Component", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it("should mount the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have a container with name resetPasswordMainContainer", () => {
    expect(wrapper.find("#resetPasswordMainContainer")).toBeTruthy();
  });

  it("checks for on resetPassword button function to be called", () => {
    const e = { preventDefault: () => {} };
    jest.spyOn(e, "preventDefault");
    wrapper.find("#resetPasswordButton").simulate("click", e);
    expect(e.preventDefault).toBeCalled();
  });

  it("checks for Sign Up button click ", () => {
    wrapper.find("#resetPasswordButton").simulate("click");
  });
});
