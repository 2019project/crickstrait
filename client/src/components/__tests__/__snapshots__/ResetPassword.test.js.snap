// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ResetPassword Component should mount the component 1`] = `
<div>
  <div
    id="resetPasswordMainContainer"
  >
    <form
      id="resetPasswordForm"
    >
      <p>
        <button
          id="resetButton"
        >
          Reset Password
        </button>
      </p>
      <p>
        <input
          id="resetPasswordInput"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          type="text"
          value=""
        />
        <span />
      </p>
      <p
        id="resetPasswordErrorMessage"
      />
      <p>
        <input
          id="resetPasswordInput"
          name="password"
          onChange={[Function]}
          placeholder="New Password"
          type="password"
          value=""
        />
      </p>
      <p
        id="resetPasswordErrorMessage"
      />
      <p>
        <input
          id="resetPasswordInput"
          name="confirmPassword"
          onChange={[Function]}
          placeholder="Confirm Password"
          type="password"
          value=""
        />
      </p>
      <p
        id="resetPasswordErrorMessage"
      />
      <p>
        <button
          id="resetPasswordButton"
          onClick={[Function]}
        >
          Proceed
        </button>
      </p>
    </form>
  </div>
</div>
`;
